10_rest/n0npax/cmd/puppy-server/main_test.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"testing"
+
+	"bou.ke/monkey"
```

Comment:
monkey patching strikes me as strange here and should be avoided if possible.

Reply:
Only one monkey patch is still here. Passing invalid port `65xxx+` causes `os.Exit`.
I may be wrong, but I believe I can't overwrite internal `gin` behavior